cli/src/stake.rs

```diff
@@ -1980,40 +1980,38 @@ pub fn process_split_stake(
     };
 
     let rent_exempt_reserve = if !sign_only {
-        if let Ok(stake_account) = rpc_client.get_account(&split_stake_account_address) {
-            if stake_account.owner == stake::program::id() {
-                return Err(CliError::BadParameter(format!(
+        let check_stake_account = |account: Account| -> Result<u64, CliError> {
+            match account.owner {
+                owner if owner == stake::program::id() => Err(CliError::BadParameter(format!(
                     "Stake account {split_stake_account_address} already exists"
```

Comment:
this isn't really a problem if the delegation is inactive, right?

Reply:
Um, do we allow splitting into an existing inactive stake account? I'll have to double check.

Reply:
Yeah, the only time a split to an account that is already owned by the stake program won't fail with a stake-program error is if the split recipient contains the state `StakeStateV2::Uninitialized`; the delegation (split source) status doesn't matter.
As there's no way to use the cli to get an account in that state, I think this logic is okay.